app/models/policy_version.rb

```diff
@@ -86,6 +86,61 @@ def before_save
     self.policy_sha256 = Digest::SHA256.hexdigest(policy_text)
   end
 
+  def after_save
+    ensure_versions_limit
+  rescue => e
+    Rails.logger.error("Error while ensuring policy '#{id}' versions retention limit: #{e.inspect}")
+  end
+
+  def ensure_versions_limit
```

Comment:
this method is redundant imo. you can implement `log_versions_off_limit` and `delete_versions_off_limit` in `after_save` and it will be clearer without letting the user figure out what this ensure is